src/DssProxyActionsCharter.sol

```diff
@@ -293,18 +330,28 @@ contract DssProxyActionsCharter is Common {
 
     function freeGem(
         address gemJoin,
+        uint256 cdp,
         uint256 amt
     ) external {
         // Unlocks token amount from the CDP
-        _frob(GemJoinLike(gemJoin).ilk(), -_toInt256(_convertTo18(gemJoin, amt)), 0);
+        _frob(
+            CdpRegistryLike(cdpRegistry).ilks(cdp),
+            CdpRegistryLike(cdpRegistry).owns(cdp),
+            -_toInt256(_convertTo18(gemJoin, amt)),
+            0
+        );
         // Exits token amount to the user's wallet as a token
         CharterLike(charter).exit(gemJoin, msg.sender, amt);
     }
 
     function exitETH(
         address ethJoin,
+        uint256 cdp,
         uint256 wad
     ) external {
+        require(CdpRegistryLike(cdpRegistry).owns(cdp) == address(this), "wrong-cdp");
```

Comment:
This makes me wonder why we didn't allow to pass a custom `allowed(u)` to the `charter.exit` function.

Reply:
If I understand you correctly, the answer is that we figured out you can always `flux()` to `address(this)` before you exit.
That is the same flow as in the cdp-manager.

Reply:
It was to match the gemJoin construction, but yes Tal is right you can just flux to achieve the same thing although it's more gas.

Reply:
Yeah ok, I was just wondering because that would remove this require, but yeah it is fine as it is.